gossipsub_spam_test.go

```diff
@@ -662,10 +662,10 @@ func TestGossipsubAttackInvalidMessageSpam(t *testing.T) {
 					// fail validation and reduce the attacker's score)
 					for i := 0; i < 100; i++ {
 						msg := &pb.Message{
-							Data:     []byte("some data" + strconv.Itoa(i)),
-							TopicIDs: []string{mytopic},
-							From:     []byte(attacker.ID()),
-							Seqno:    []byte{byte(i + 1)},
+							Data:  []byte("some data" + strconv.Itoa(i)),
+							Topic: &mytopic,
```

Comment:
I think we should remove the `optional` trait from the protobuf file for the topic. It is compulsory (AFAIK) and it would remove this annoying indirection, and speed things up slightly.

Reply:
Agreed. But it's a bit of work, so let's do it in a future pr.